agent/engine/docker_task_engine.go

```diff
@@ -2666,8 +2669,8 @@ func (engine *DockerTaskEngine) removeContainer(task *apitask.Task, container *a
 
 // updateTaskUnsafe determines if a new transition needs to be applied to the
 // referenced task, and if needed applies it. It should not be called anywhere
-// but from 'AddTask' and is protected by the tasksLock lock there.
-func (engine *DockerTaskEngine) updateTaskUnsafe(task *apitask.Task, update *apitask.Task) {
+// but from 'UpdateTask' and is protected by the tasksLock lock there.
+func (engine *DockerTaskEngine) updateTaskUnsafe(task *apitask.Task, updateDesiredStatus apitaskstatus.TaskStatus) {
```

Comment:
nit: maybe we should rename this method to `updateTaskDesiredStateUnsafe` to make it even more clearer.

Reply:
Good idea, will make this change in the following revision.